packages/hooks/src/test/index.test.js

```diff
@@ -620,3 +620,58 @@ test( 'Test `this` context via composition', () => {
 	const testObject2 = {};
 	Object.assign( testObject2, createHooks() );
 } );
+
+test( 'adding a hook triggers a hookAdded action passing all callback details', () => {
+	const hook_added_spy = jest.fn();
+
+	// Set up a listener for the 'hookAdded' action.
+	addAction( 'hookAdded', 'my_callback', hook_added_spy );
+
+	// Test addAction.
+	addAction( 'testAction', 'my_callback2', action_a, 9 );
+	expect( hook_added_spy ).toHaveBeenCalledTimes( 1 );
+	expect( hook_added_spy ).toHaveBeenCalledWith(
+		'testAction',
+		'my_callback2',
+		action_a,
+		9
+	);
+
```

Comment:
Personally, I would split this and the following test into 2 parts to make them more focused. However, it's not mandatory.

Reply:
Yea, making the tests more granular like this is nicer - thanks!